pkg/start/start.go

```diff
@@ -154,14 +150,14 @@ func (o *Options) Run() error {
 }
 
 func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourcelock.ConfigMapLock) {
-	// listen on metrics
-	if len(o.ListenAddr) > 0 {
-		mux := http.NewServeMux()
-		mux.Handle("/metrics", promhttp.Handler())
+	runContext, runCancel := context.WithCancel(ctx)
+	shutdownContext, shutdownCancel := context.WithCancel(ctx)
```

Comment:
`go vet` complains about paths with unused contexts
I'm not too worried about it, but it seems
```
	defer func() {
		runCancel()
		shutdownCancel()
	}()
```
after this declaration would fix that. WDYT?

Reply:
Done with ab8e70d -> b0896ae, although I used two separate `defer`, on the off chance that the second `WithCancel` panics and some higher level function catches the panic and decides to keep going anyway ;).